docs/reference/terminology.md

```diff
@@ -1,29 +1,951 @@
-<!--
- Copyright 2024 Google LLC
+# Terminology
 
- Licensed under the Apache License, Version 2.0 (the "License");
- you may not use this file except in compliance with the License.
- You may obtain a copy of the License at
+This page provides definitions for technical terms and concepts used throughout the MaxText codebase and documentation. Terms are organized by category for easy reference.
 
-      https://www.apache.org/licenses/LICENSE-2.0
+## Table of Contents
 
- Unless required by applicable law or agreed to in writing, software
- distributed under the License is distributed on an "AS IS" BASIS,
- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- See the License for the specific language governing permissions and
- limitations under the License.
- -->
+- [Performance Metrics](#performance-metrics)
+- [Hardware & Interconnect](#hardware--interconnect)
+- [Parallelism Strategies](#parallelism-strategies)
+- [Attention Mechanisms](#attention-mechanisms)
+- [Positional Encodings](#positional-encodings)
+- [Model Architecture](#model-architecture)
+- [Mixture of Experts (MoE)](#mixture-of-experts-moe)
+- [Quantization](#quantization)
+- [Memory Optimization](#memory-optimization)
+- [Checkpointing](#checkpointing)
+- [JAX & XLA Concepts](#jax--xla-concepts)
+- [Training & Optimization](#training--optimization)
+- [Data Pipeline](#data-pipeline)
+- [Inference](#inference)
+- [Configuration & Infrastructure](#configuration--infrastructure)
 
-# Terminology
+---
+
+## Performance Metrics
+
+**FLOP (Floating Point Operation):**
+A single floating-point arithmetic operation (addition, multiplication, etc.).
+
+**FLOPS (Floating Point Operations):**
+Plural form of FLOP.
+
+**FLOP/s or FLOPs:**
```

Comment:
I don't recall we have `FLOPs` to be `per second`, could you help remove (just keep FLOP/s) or cross check?